lib/clerk/rack_middleware_v2.rb

```diff
@@ -170,8 +170,19 @@ def call(env)
         return unknown(interstitial: true)
       end
 
-      if production? && client_uat.nil?
-        return signed_out(env)
+      # Show interstitial when there is no client_uat or cookie token
+      if (client_uat == "0" || client_uat.nil?) && cookie_token.to_s.empty?
+        return unknown(interstitial: true)
+      end
+
+      # Show interstitial when there is client_uat is incompatible with cookie token
+      has_cookie_token_without_client = (client_uat == "0" || client_uat.nil?) && cookie_token
+      has_client_without_cookie_token = (client_uat.to_s != "0" || client_uat.to_s != "") && cookie_token.to_s.empty?
+      return unknown(interstitial: true) if has_cookie_token_without_client || has_client_without_cookie_token
+
+      # Show interstitial when there is client_uat is incompatible with cookie token
+      if (client_uat == "0" || client_uat.nil?) && cookie_token.to_s.empty?
```

Comment:
🔧 Same check as in line 174 could be assigned to a variable? Just for aggregating the cases at a single location maybe?